util/generate_ticker_file.py

```diff
@@ -0,0 +1,173 @@
+    Removes all instances of a particular entity from a list
+  
+    >>> remove_entity_from_list(['test', 'test', 'three'], 1)
+    ['three']
+    """
+    return [x for x in unique_list if x != unique_list[i]]
+
+
+def patch_entity_list_duplicates(unique_list, original_list, i):
+    """
+    Updates a list containing duplicates at index i with longer entities from the original list
+  
+    >>> patch_entity_list_duplicates(['test', 'test', 'three'], ['test one', 'test two', 'three'], 1)
+    ['test one', 'test two', 'three']
+    >>> patch_entity_list_duplicates(['test', 'test', 'three'], ['test', 'test', 'three'], 1)
+    WARNING: Could not remove duplicate test with entity ['test']
+    WARNING: Could not remove duplicate test with entity ['test']
+    ERROR: More than one duplicate could not be resolved at entity 2 of 3
+    ['test', 'test', 'three']
+    >>> patch_entity_list_duplicates(['test', 'test', 'three'], ['test', 'test one', 'three'], 1)
+    WARNING: Could not remove duplicate test with entity ['test']
+    ['test', 'test one', 'three']
+    >>> patch_entity_list_duplicates(
+    ... ['a', 'china', 'b', 'china', 'c', 'china'],
+    ... ['a', 'china sxt pharma', 'b', 'china kempei', 'c', 'china inc'], 1)
+    ['a', 'china sxt', 'b', 'china kempei', 'c', 'china inc']
+    """
+    current_entity = unique_list[i]
+    indices_to_change = [
+        index for index in range(len(unique_list))
+        if unique_list[index] == current_entity
+    ]
+
+    possible_conflicts = []
+    for index in indices_to_change:
+        original_entity = original_list[index].split(' ')
+
+        if len(original_entity) < len(current_entity.split(' ')) + 1:
+            print("WARNING: Could not remove duplicate {0} with entity {1}".format(
+                unique_list[i], original_entity))
+            possible_conflicts.append(index)
+        else:
+            unique_list[index] = ' '.join(
+                original_entity[:len(current_entity.split(' ')) + 1])
+
+    if len(possible_conflicts) > 1:
+        print(
+            "ERROR: More than one duplicate could not be resolved at entity {0} of {1}".
+            format(i + 1, len(original_list)))
+
+    return unique_list
+
+
+def update_entity_working_list(original_entities, seen_entities, entity_working_list, i):
+    if entity_working_list[i] in seen_entities:
+        first_duplicate = seen_entities.index(entity_working_list[i])
+        entity_working_list = patch_entity_list_duplicates(entity_working_list,
+                                                           original_entities, i)
+        seen_entities[first_duplicate] = entity_working_list[first_duplicate]
+
+    return seen_entities, entity_working_list
+
+
+def create_unique_minimum_entities(entities):
+    """
+    Retuns an entity list where each entity is as short
+    as possible while maintaining a unique list
+  
+    >>> create_unique_minimum_entities(['abc one company', 'abc two company', 'def three company'])
```

Comment:
This behavior may be a problem if a client wants to add "just one more" and doesn't use the whole list. We'll need to coach clients through this one.